File: workers/worker-jvm/src/main/java/com/openregex/WorkerRunner.java
package com.openregex;

import redis.clients.jedis.Jedis;

public class WorkerRunner {
    public static void main(String[] args) {
        String redisUrl = System.getenv("REDIS_URL");
        if (redisUrl == null || redisUrl.isEmpty()) {
            redisUrl = "redis://redis:6379";
        }

        try (Jedis jedis = new Jedis(redisUrl)) {
            Registry.registerEngines(jedis);
        } catch (Exception e) {
            System.err.println("Failed to register engines: " + e.getMessage());
        }

        Processor.listenAndProcess(redisUrl);
    }
}

File: workers/worker-php/src/worker_runner.php
<?php

require __DIR__ . '/../vendor/autoload.php';

use OpenRegex\Worker\Registry;
use OpenRegex\Worker\Processor;

$redisUrl = getenv('REDIS_URL') ?: 'redis://redis:6379';
$client = new Predis\Client($redisUrl);

try {
    Registry::registerEngines($client);
    Processor::listenAndProcess($client);
} catch (\Exception $e) {
    echo "Fatal error during startup: " . $e->getMessage() . "\n";
    exit(1);
}